app/src/main/java/com/duckduckgo/app/di/SystemComponentsModule.kt

```diff
@@ -47,4 +50,9 @@ open class SystemComponentsModule {
     @Provides
     fun appIconModifier(context: Context, appShortcutCreator: AppShortcutCreator): IconModifier =
         AppIconModifier(context, appShortcutCreator)
+
+    @Provides
```

Comment:
This seems to be tied to the app lifecycle, just wondering if it makes sense for it to be a Singleton?

Reply:
Because it's stateless and part of the application component, I usually tend to leave out the `@singleton`. Just micro-optimization in dagger that can avoid some checks when I know it's not really important to be singleton, and knowing how I will use it in our code too. So, it's same result at the end.